lib/suse/connect/api.rb

```diff
@@ -156,6 +156,29 @@ def system_activations(auth)
         @connection.get('/connect/systems/activations', :auth => auth)
       end
 
+      # Lists all available upgrade paths for a given list of products
+      #
+      # @param auth [String] authorization string which will be injected in 'Authorization' header in request.
+      #   In this case we expect Base64 encoded string with login and password
+      # @param [Array <Remote::Product>] a list of producs
+      #
+      # @return [Array <Array <Remote::Product>>] the list of possible upgrade paths for the given products
+      def system_migrations(auth, products)
+        products_attributes = products.map {|product| serialize_product(product) }
```

Comment:
Let's move products_attributes (def attributes or to_aatributes or to_params) to SUSE::Connect::Remote::Product